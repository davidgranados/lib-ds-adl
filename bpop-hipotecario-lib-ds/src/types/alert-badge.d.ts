import { ReactNode } from "react";
import { Testable } from "./global";

export interface AlertBadgeProps extends Testable {
  children?: ReactNode;
  type?: "info" | "warning" | "success" | "danger";
}
