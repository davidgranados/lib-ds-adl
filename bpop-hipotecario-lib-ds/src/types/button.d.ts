import { HTMLPropsFor } from "./component";

export type ButtonVariant = "primary" | "secondary";
export type ButtonTextSize = "sm" | "md" | "lg";
export type ButtonSize = "xs" | ButtonTextSize;

export interface DefaultButtonProps extends HTMLPropsFor<"button"> {
  label: string;
  variant?: ButtonVariant;
  disabled?: boolean;
  onClick?: () => void;
}

export interface ButtonProps extends DefaultButtonProps {
  size?: ButtonSize;
}

export interface ButtonTextProps extends DefaultButtonProps {
  size?: ButtonTextSize;
}
