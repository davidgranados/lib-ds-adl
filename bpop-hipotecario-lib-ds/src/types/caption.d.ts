import { ReactNode } from "react";
import { Testable } from "./global";

export declare type InputFieldCaptionStatus = "default" | "error" | "warning" | "success";

export declare interface InputFieldCaptionProps extends Testable {
  children: ReactNode;
  status?: InputFieldCaptionStatus;
}
