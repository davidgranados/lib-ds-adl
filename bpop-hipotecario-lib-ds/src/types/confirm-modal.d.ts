import { Testable } from "./global";

export declare interface ConfirmModalProps extends Testable {
  title: string;
  description?: string;
  show: boolean;
  mainButtonText?: string;
  secondaryButtonText?: string;
  onMainButtonClick?: () => void;
  onSecondaryButtonClick?: () => void;
  onToggle?: (state: boolean) => void;
}
