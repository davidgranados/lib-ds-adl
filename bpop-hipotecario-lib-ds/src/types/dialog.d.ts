import { Testable } from "./global";

export declare interface DialogProps extends Testable {
  show: boolean;
  width?: number;
  onToggle?: (state: boolean) => void;
}
