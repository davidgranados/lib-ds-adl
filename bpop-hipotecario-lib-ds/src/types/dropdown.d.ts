import { Testable } from "./global";

export declare interface DropdownProps extends Testable {
  label: string;
  items: { label: string; value: string | number }[];
  wrapperId?: string;
  wrapperDataTestId?: string;
  labelId?: string;
  labelDataTestId?: string;
  captionId?: string;
  captionDataTestId?: string;
  disabled?: boolean;
  hasError?: boolean;
  caption?: string;
  placeholder?: string;
  value?: string | number;
  onChange: (v: string | number) => void;
}
