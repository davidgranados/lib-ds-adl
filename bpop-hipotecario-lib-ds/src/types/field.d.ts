import { ChangeEventHandler, FocusEventHandler, FormEventHandler } from "react";

import { InputBaseSize, InputBaseStatus } from "../../atoms/input-base";
import { TypographyStatus } from "../../atoms/typography";

export interface DefaultFieldProps {
  id: string;
  label: string;
  value?: string | number;
  status?: InputBaseStatus | TypographyStatus;
  caption?: string;
  placeholder?: string;
  size?: InputBaseSize;
  disabled?: boolean;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  onFocus?: FocusEventHandler<HTMLInputElement>;
  onBlur?: FocusEventHandler<HTMLInputElement>;
  onInput?: FormEventHandler<HTMLInputElement>;
}
