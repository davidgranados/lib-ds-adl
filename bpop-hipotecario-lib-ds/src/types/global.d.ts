export declare interface Testable {
  id?: string;
  dataTestId?: string;
}
