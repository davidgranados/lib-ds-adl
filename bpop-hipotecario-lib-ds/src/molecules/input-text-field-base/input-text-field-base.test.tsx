import React from "react";
import { mount, ReactWrapper } from "enzyme";

import { InputBaseSize, InputFieldType } from "../../types";

import InputTextFieldBase from "./input-text-field-base";

describe("<InputTextFieldBase />", () => {
  let wrapper: ReactWrapper<typeof InputTextFieldBase>;
  const type: InputFieldType = "text";
  const size: InputBaseSize = "md";
  const caption = "test caption";
  const props = {
    id: "test-number-field-old-id",
    label: "test number field input-field-label",
    type: type,
    maxLength: 10,
    placeholder: "test placeholder",
    size: size,
    validateInputValue: jest.fn(),
    onFocus: jest.fn(),
    onBlur: jest.fn(),
    onChange: jest.fn(),
    onInput: jest.fn(),
  };
  describe("with initial value and caption", () => {
    const initialValue = "initial value";
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof InputTextFieldBase>(
        <InputTextFieldBase {...props} value={initialValue} caption={caption} />
      );
    });
    it("should match snapshot", () => {
      // then
      expect(wrapper).toMatchSnapshot();
    });
    describe("<input-field-label />", () => {
      it("should render correctly", () => {
        // when
        const labelElement = wrapper.find("label");
        // then
        expect(labelElement.exists()).toBe(true);
        expect(labelElement.prop("htmlFor")).toEqual(props.id);
        expect(labelElement.prop("children")).toEqual(props.label);
      });
    });
    describe("<InputTextBase />", () => {
      it("should render correctly", () => {
        // when
        const inputElement = wrapper.find("input");
        // then
        expect(inputElement.exists()).toBe(true);
        expect(inputElement.prop("id")).toEqual(props.id);
        expect(inputElement.prop("value")).toEqual(initialValue);
        expect(inputElement.prop("type")).toEqual(type);
        expect(inputElement.prop("placeholder")).toEqual(props.placeholder);
        expect(inputElement.hasClass(`size-${props.size}`)).toBe(true);
        expect(inputElement.hasClass(`filled`)).toBe(true);
      });
    });
    describe("caption <span />", () => {
      it("should render correctly", () => {
        // when
        const spanElement = wrapper.find("span");
        // then
        expect(spanElement.exists()).toBe(true);
        expect(spanElement.prop("children")).toEqual(caption);
      });
    });
  });
  describe("without initial value or caption", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof InputTextFieldBase>(<InputTextFieldBase {...props} />);
    });
    it("should match snapshot", () => {
      // then
      expect(wrapper).toMatchSnapshot();
    });
    describe("<InputTextBase />", () => {
      it("should render correctly", () => {
        // when
        const inputElement = wrapper.find("input");
        // then
        expect(inputElement.exists()).toBe(true);
        expect(inputElement.prop("value")).toEqual("");
        expect(inputElement.hasClass(`filled`)).toBe(false);
      });
    });
    describe("caption <span />", () => {
      it("shouldn't render", () => {
        // when
        const spanElement = wrapper.find("span");
        // then
        expect(spanElement.exists()).toBe(false);
      });
    });
  });
  describe("with error", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof InputTextFieldBase>(<InputTextFieldBase {...props} hasError={true} caption={caption} />);
    });
    it("should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
    describe("<InputTextBase />", () => {
      it("should have error class", () => {
        // when
        const inputElement = wrapper.find("input");
        // then
        expect(inputElement.exists()).toBe(true);
        expect(inputElement.hasClass("error")).toBe(true);
      });
    });
    describe("caption <span />", () => {
      it("should render correctly", () => {
        // when
        const spanElement = wrapper.find("span");
        // then
        expect(spanElement.exists()).toBe(true);
        expect(spanElement.hasClass("error")).toBe(true);
        expect(spanElement.prop("children")).toEqual(caption);
      });
    });
  });
  describe("change event", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof InputTextFieldBase>(<InputTextFieldBase {...props} />);
    });
    it("should handle change", () => {
      // given
      const mockChangeEvent = { target: { value: "test" } };
      // when
      wrapper.find("input").simulate("change", mockChangeEvent);
      // then
      expect(props.validateInputValue).toHaveBeenCalled();
      expect(props.onChange).toHaveBeenCalled();
    });
  });
});
