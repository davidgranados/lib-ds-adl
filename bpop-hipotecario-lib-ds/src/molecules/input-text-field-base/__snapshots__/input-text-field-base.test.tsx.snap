// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputTextFieldBase /> with error should match snapshot 1`] = `
<div
  data-testid="test-number-field-old-id-wrapper"
  id="test-number-field-old-id-wrapper"
>
  <label
    className="default"
    data-testid="test-number-field-old-id-label"
    htmlFor="test-number-field-old-id"
    id="test-number-field-old-id-label"
  >
    test number field input-field-label
  </label>
  <input
    className="default size-md error"
    data-testid="test-number-field-old-id"
    id="test-number-field-old-id"
    maxLength={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[MockFunction]}
    onInput={[Function]}
    placeholder="test placeholder"
    type="text"
    value=""
  />
  <span
    className="default error"
    data-testid="test-number-field-old-id-caption"
    id="test-number-field-old-id-caption"
  >
    test caption
  </span>
</div>
`;

exports[`<InputTextFieldBase /> with initial value and caption should match snapshot 1`] = `
<div
  data-testid="test-number-field-old-id-wrapper"
  id="test-number-field-old-id-wrapper"
>
  <label
    className="default"
    data-testid="test-number-field-old-id-label"
    htmlFor="test-number-field-old-id"
    id="test-number-field-old-id-label"
  >
    test number field input-field-label
  </label>
  <input
    className="default size-md filled"
    data-testid="test-number-field-old-id"
    id="test-number-field-old-id"
    maxLength={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[MockFunction]}
    onInput={[Function]}
    placeholder="test placeholder"
    type="text"
    value="initial value"
  />
  <span
    className="default"
    data-testid="test-number-field-old-id-caption"
    id="test-number-field-old-id-caption"
  >
    test caption
  </span>
</div>
`;

exports[`<InputTextFieldBase /> without initial value or caption should match snapshot 1`] = `
<div
  data-testid="test-number-field-old-id-wrapper"
  id="test-number-field-old-id-wrapper"
>
  <label
    className="default"
    data-testid="test-number-field-old-id-label"
    htmlFor="test-number-field-old-id"
    id="test-number-field-old-id-label"
  >
    test number field input-field-label
  </label>
  <input
    className="default size-md"
    data-testid="test-number-field-old-id"
    id="test-number-field-old-id"
    maxLength={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[MockFunction]}
    onInput={[Function]}
    placeholder="test placeholder"
    type="text"
    value=""
  />
</div>
`;
